instrumentation/internal/internal-class-loader/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/internal/classloader/BootDelegationInstrumentation.java

```diff
@@ -47,10 +47,14 @@ public class BootDelegationInstrumentation implements TypeInstrumentation {
   public ElementMatcher<TypeDescription> typeMatcher() {
     // just an optimization to exclude common class loaders that are known to delegate to the
     // bootstrap loader (or happen to _be_ the bootstrap loader)
+    // The AgentClassLoader and InstrumentationModuleClassloaders are required to be excluded
+    // for the instrumentation to work properly. loadClass on those is invoked during Advice
+    // bootstrapping which therefore would cause an infinite recursion
     return not(namedOneOf(
             "java.lang.ClassLoader",
             "com.ibm.oti.vm.BootstrapClassLoader",
-            "io.opentelemetry.javaagent.bootstrap.AgentClassLoader"))
+            "io.opentelemetry.javaagent.bootstrap.AgentClassLoader",
+            "io.opentelemetry.javaagent.tooling.instrumentation.indy.InstrumentationModuleClassLoader"))
```

Comment:
Is this really required, won't https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/6d94506b795e5c720758839005505a428217cdae/javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/ignore/GlobalIgnoredTypesConfigurer.java#L63 already exclude the `AgentClassLoader` and I'd guess that `InstrumentationModuleClassLoader` is also already excluded because it is loaded by `AgentClassLoader`.

Reply:
You are right, this is not required here. I'll remove it.